render/query.go

```diff
@@ -218,11 +218,24 @@ func (r *Reply) getDataAggregated(ctx context.Context, cfg *config.Config, tf Ti
 	data = EmptyData
 
 	metricList := targets.AM.Series(targets.isReverse)
-
 	if len(metricList) == 0 {
 		return
 	}
 
+	var metricListUnreverse []string
+	if targets.isReverse {
+		metricListUnreverse = targets.AM.Series(false)
+	} else {
+		metricListUnreverse = metricList
+	}
+
+	var metricListRuleLookup []string
+	if targets.isReverse && targets.rollupUseReverted {
+		metricListRuleLookup = metricListUnreverse
+	} else {
+		metricListRuleLookup = metricList
+	}
+
```

Comment:
We need only one additional slice
```suggestion
	var metricListRuleLookup []string
	if targets.isReverse {
		metricListRuleLookup = targets.AM.Series(true)
	} else {
		metricListRuleLookup = metricList
	}

```

Reply:
> We need only one additional slice

It's not so simple.  

We need reverse or unreverse options  on some cases, like
`for n, m := range metricListRuleLookup {` - m is used for roollup lookup

We need unreverse  on some cases, like (for store in steps)
```
if targets.isReverse {
	m = metricListUnreverse[n]
}
```